### PR TITLE
Helpers for writing raw HTTP body; handle test cases with raw request/response in test runnerf

### PR DESCRIPTION
**Pull request by jhump**
This is a preliminary change that adds a couple of `internal` helpers that the reference client and server will use and to add a little validation to the test runner related to the use of raw requests and responses.

### CODE REVIEW

File: internal/app/connectconformance/server_runner.go [hunk 1]
```diff
@@ -176,7 +176,7 @@ func runTestCasesForServer(
 				results.assert(name, expectations[resp.TestName], resp.GetResponse())
 			}
 			if isReferenceClient {
-				for _, msg := range resp.Feedback {
+				for _, msg := range resp.GetFeedback() {
```
Comment (jhump) [inline internal/app/connectconformance/server_runner.go:179]: This is an unrelated fix. I encountered a nil-pointer-defereference here at one point while developing this arc of work in #682. Since this PR has other changes in the test runner, seemed okay to drop this here.

File: internal/headers.go [hunk 2]
```diff
@@ -32,6 +32,19 @@ func AddHeaders(
 	}
 }
 
+// AddTrailers adds all header values in src to dest, but
+// it prefixes each header name with http.TrailerPrefix.
```
Comment (jhump) [inline internal/headers.go:36]: This is how to add trailers from a server handler with `net/http`.

**Review by smaye81 — APPROVED** [review]
